### DISCUSSION TITLE
Add support for vim-style key bindings

### DISCUSSION THREAD

**Discussion by BrendanMesters** [d99kris/nchat#241] [Ideas]
I would be interested in having a simplified mode based control system for nchat, similar to vi.

My thoughts on the user end experience would be that you'll be able to access the three different panes (chats list, current chat, current message) from _"normal mode"_ via ctrl+w+hjkl pane switching commands, move around in the chat lists and current chat window via jk and to have the different special functions (such as reply, edit, the emoji menu, etc) bound to single key commands in normal mode.

This should only be an optional feature, enabled in the `key.conf` file (e.g. vi_mode=0 by default, vi_mode=1 if you want to enable it). 

If this is something which the nchat community would like then I'd be happy to try and implement that sometime :-)

**Comment by loeffel-io** [comment]
+100

**Comment by tucama** [comment]
I love this

**Comment by EugenioBertolini** [comment]
It would be so good

**Comment by mashtonian** [comment]
Yes please to this!

**Comment by abdenourgnx** [comment]
Please 🔥

**Comment by gensimone** [comment]
That would be great!

**Comment by oscarmlage** [comment]
I'd love to have  this feat!

**Comment by nasteffe** [comment]
If the energy required to do so is not prohibitive, I would love to see this feature added! While I understand that most users would not know vim style keybindings, I do imagine having basic ones would actually make navigating the different spaces much easier and would certainly make the simple keybindings easier to learn and keep track of. 

Edited for context.

**Comment by pinusc** [comment]
I have no idea how feasible this feature is, but something that might be worth considering is to "just" use [readline](https://tiswww.case.edu/php/chet/readline/rltop.html) to handle the textbox, and that just comes with automagic vim support. 

This would only add vim keybindings in the edit box, but I'm not entirely sure how useful it would be to have them outside...

**Comment by semedilino-073** [comment]
I think this would easily turn `nchat` into the best Whatsapp Client, which it is already, but not always :)

**Comment by mohammadazeemwani** [comment]
Pleaaaaaseee add this.
![besos](https://github.com/user-attachments/assets/a9ba2ce9-8c92-458a-b64f-a431ca0a5a2d)

**Comment by jonnieey** [comment]
💯💯💯 👍👍👍

**Comment by ijuttt** [comment]
just here to add +1, would be great

**Comment by Frankwii** [comment]
Please, this would be amazing!!

**Comment by d99kris** [comment]
A proposed set of key bindings has been shared by a contributor in #535 - for those with good vim familiarity, and ideally also with other apps providing vim-binding controls - feel free to share input on the proposal. I will also study the same and review it, but it's good with additional perspective.

**Comment by loeffel-io** [comment]
i am back since prebuilt binaries are supported. tried to configure the keys but it just feel so wrong. its not even about the input window. i would open the external editor for this anyways but the moves are just so weird 

also would love to move with ctrl+h/j/k/l etc like in neovim which does not work; setting the keys to h/j/k/l only just breaks the chat search etc

**Comment by desgua** [comment]
I'm working on a vim style navigation for the user interface. I'd appreciate help on testing and insights. 
I made a pull request here: [567](https://github.com/d99kris/nchat/pull/567)

**Reply by desgua** [reply]
I close the PR but the implementation can be seen [here](https://github.com/desgua/nchat/tree/add_vim_style_navigation_for_ui).

**Comment by Luquatic** [comment]
any progress? I'm really missing this feature

**Reply by desgua** [reply]
If you are looking for a vim style to navigate the user interface, check [desgua fork](https://github.com/desgua/nchat). If you want the vim style for the compose entry, check the [543 PR](https://github.com/d99kris/nchat/pull/543).
